File: api/Helpers/Params/UserParams.cs
namespace api.Helpers.Params
{
    public class UserParams : PaginationParams
    {
        public string Username { get; set; }
        public string Gender { get; set; }
        public int MinAge { get; set; } = 18;
        public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";

    }


}

File: client/src/app/_models/userParams.ts
import { UserModel } from "./userModel";

export class UserParams {
    gender : string;
    minAge = 18;
    maxAge = 99;
    pageNumber = 1;
    pageSize = 12;
    orderBy = "lastActive";

    constructor(user : UserModel){
        this.gender = user.gender === 'female' ? 'male' : 'female'
    }
}